Tribler/Core/Video/VideoServer.py

```diff
@@ -33,6 +39,47 @@ def __init__(self, port, session, video_player):
         self.daemon_threads = True
         self.allow_reuse_address = True
 
+    def get_vod_download(self):
+        """
+        Return the current Video-On-Demand download that is being requested.
+        """
+        return self.vod_download
+
+    def set_vod_download(self, new_download):
+        """
+        Set a new Video-On-Demand download. Set the mode of old download to normal and close the file stream of
+        the old download.
+        """
+        if self.vod_download:
+            self.vod_download.set_mode(DLMODE_NORMAL)
+            vi_dict = self.vod_info.pop(self.vod_download.get_def().get_infohash(), None)
+            if vi_dict and 'stream' in vi_dict:
+                vi_dict['stream'][0].close()
+
+        self.vod_download = new_download
+
+    def get_vod_stream(self, dl_hash, wait=False):
+        if 'stream' not in self.vod_info[dl_hash] and self.session.get_download(dl_hash):
+            download = self.session.get_download(dl_hash)
+            vod_filename = self.get_vod_destination(download)
+            while wait and not os.path.exists(vod_filename):
+                time.sleep(1)
```

Comment:
This code is moved from `VideoPlayer.py` and I did not adjust the logic. I can refactor the video player to use deferreds but I don't want to do too much at once.